src/points/saga.ts

```diff
@@ -0,0 +1,61 @@
+import { call, put, select, spawn, takeLeading } from 'typed-redux-saga'
+import { getHistoryStarted, getHistorySucceeded, getHistoryError } from 'src/points/slice'
+import { safely } from 'src/utils/safely'
+import networkConfig from 'src/web3/networkConfig'
+import { GetHistoryResponse } from 'src/points/types'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { getPointsHistoryNextPageUrlSelector } from 'src/points/selectors'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'Points/saga'
+
+export async function fetchHistory(
+  address: string,
+  url?: string | null
+): Promise<GetHistoryResponse> {
+  const response = await fetchWithTimeout(
+    url ?? `${networkConfig.getPointsHistoryUrl}?` + new URLSearchParams({ address }),
+    {
+      method: 'GET',
+    }
+  )
+  if (response.ok) {
+    return response.json() as Promise<GetHistoryResponse>
+  } else {
+    throw new Error(await response.text())
+  }
+}
+
+export function* getHistory({ payload: params }: ReturnType<typeof getHistoryStarted>) {
```

Comment:
The idea here is that we'll emit `getHistoryStarted` with `fromPage: false` on all opens of the history bottom sheet. On non-first-time-opens, this will have the effect of rendering the old Redux state + loading indicator while we fetch the new data, and the new data will overwrite the old data (see my other comment for more thoughts on this). When a user drags down to fetch more data, set `fromPage: true`, which will fetch using the page params stored in Redux, and _append_ the results to Redux state rather than overwriting them.

Reply:
this makes sense, essentially we're using each bottom sheet load to clear and refresh the cache. i think that works well, although it makes me even more convinced that we should do a background fetch on the points home screen mount rather than on bottom sheet open, since the user could close and reopen the bottom sheet without leaving the screen. wdyt?

can we rename `fromPage` to `fetchFirstPage` / `refreshCache` / `getNextPage` or similar? it was hard to understand what `fromPage` meant at first